read-fonts/src/tables/postscript/charstring.rs

```diff
@@ -1,203 +1,644 @@
 //! Parsing for PostScript charstrings.
 
-use super::{Error, Number};
-use crate::{types::Fixed, Cursor};
+use super::{BlendState, Error, Index, Stack};
+use crate::types::{Fixed, Pen};
 
-/// PostScript charstring operator.
+/// Maximum nesting depth for subroutine calls.
 ///
-/// See <https://learn.microsoft.com/en-us/typography/opentype/spec/cff2charstr#appendix-a-cff2-charstring-command-codes>
-// TODO: This is currently missing legacy math and logical operators.
-// fonttools doesn't even implement these: <https://github.com/fonttools/fonttools/blob/65598197c8afd415781f6667a7fb647c2c987fff/Lib/fontTools/misc/psCharStrings.py#L409>
-#[derive(Copy, Clone, PartialEq, Eq, Debug)]
-pub enum Operator {
-    HStem,
-    VStem,
-    VMoveTo,
-    RLineTo,
-    HLineTo,
-    VLineTo,
-    RrCurveTo,
-    CallSubr,
-    Return,
-    EndChar,
-    VariationStoreIndex,
-    Blend,
-    HStemHm,
-    HintMask,
-    CntrMask,
-    RMoveTo,
-    HMoveTo,
-    VStemHm,
-    RCurveLine,
-    RLineCurve,
-    VvCurveTo,
-    HhCurveTo,
-    CallGsubr,
-    VhCurveTo,
-    HvCurveTo,
-    HFlex,
-    Flex,
-    HFlex1,
-    Flex1,
+/// See "Appendix B Type 2 Charstring Implementation Limits" in
+/// <https://adobe-type-tools.github.io/font-tech-notes/pdfs/5177.Type2.pdf>
+pub const NESTING_DEPTH_LIMIT: u32 = 10;
+
+/// Trait for processing commands resulting from charstring evaluation.
+#[allow(unused_variables)]
```

Comment:
can we link to some sort of list of these commands and their definitions? See page 42 of some adobe pdf?

Reply:
Done